addons/Fractions/src/presenter.js

```diff
@@ -1145,15 +1173,32 @@ function AddonFractions_create(){
         presenter.eventBus.sendEvent('ValueChanged', eventData);
     };
 
+    presenter.onEventReceived = function (eventName, data) {
+        switch (eventName) {
+            case "ShowAnswers":
+                presenter.showAnswers();
+                break;
+            case "HideAnswers":
+                presenter.hideAnswers();
+                break;
+            case "GradualShowAnswers":
+                presenter.isGradualShowAnswersActive = true;

```

Comment:
zmienna będzie ustawiona nawet, gdy funkcja gradualShowAnswer() się nie wykona

Reply:
Zgadza się. Wzorowałem się na innych addonach, które ustawiają presenter.isGradualShowAnswersActive nawet jeżeli w samym addonie nie wszedł stan GSA dla tego addonu.

Reply:
Zazwyczaj jest to zapisywane jako 
if (!presenter.isGradualShowAnswersActive) {
    presenter.isGradualShowAnswersActive = true;
}
Jednak jestem zdania, że if jest tu niepotrzebny. Efekt końcowy będzie identyczny, zaś przypisanie jest łatwiejsze do zrozumienia

Reply:
Rozmawialiśmy na temat ten wewnętrznie i ustawienie zmiennej będzie wewnątrz metody